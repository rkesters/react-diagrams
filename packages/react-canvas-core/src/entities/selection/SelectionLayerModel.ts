import { LayerModel } from '../layer/LayerModel';
import { FactoryBank } from '../../core/FactoryBank';
import { AbstractModelFactory } from '../../core/AbstractModelFactory';
import { BaseModel } from '../../core-models/BaseModel';
import { IClientRect } from '../../states/SelectionBoxState';

export class SelectionLayerModel extends LayerModel {
	public box: IClientRect | undefined;

	constructor() {
		super({
			transformed: false,
			isSvg: false,
			type: 'selection'
		});
	}

	setBox(rect: IClientRect) {
		this.box = rect;
	}

	getChildModelFactoryBank(): FactoryBank<AbstractModelFactory<BaseModel>> | null {
		// is not used as it doesnt serialize
		return null;
	}
}
