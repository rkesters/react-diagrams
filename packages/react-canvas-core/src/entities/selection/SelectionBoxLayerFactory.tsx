import * as React from 'react';
import { AbstractReactFactory, GenerateWidgetEvent } from '../../core/AbstractReactFactory';
import { SelectionLayerModel } from './SelectionLayerModel';
import { GenerateModelEvent } from '../../core/AbstractModelFactory';
import { SelectionBoxWidget } from './SelectionBoxWidget';
import { CanvasEngine } from '../..';

export class SelectionBoxLayerFactory extends AbstractReactFactory<SelectionLayerModel> {
	constructor(engine: CanvasEngine) {
		super('selection', engine);
	}

	generateModel(event: GenerateModelEvent): SelectionLayerModel {
		return new SelectionLayerModel();
	}

	generateReactWidget(event: GenerateWidgetEvent<SelectionLayerModel>): JSX.Element {
		return <SelectionBoxWidget rect={event.model.box} />;
	}
}
