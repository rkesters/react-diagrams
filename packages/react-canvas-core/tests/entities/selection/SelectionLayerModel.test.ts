
import { SelectionLayerModel } from '../../../src/entities/selection/SelectionLayerModel';


describe('SelectionLayerModel', () => {


	test('Constructor', () => {
		const model = new SelectionLayerModel();

		expect(model.getOptions()).toMatchSnapshot({id:expect.any(String)});

		expect(model.getChildModelFactoryBank()).toBeNull();

		const rect: DOMRect = {x: 10, y: 20, top: 20, right: 50, bottom: 70, left: 10,  width: 40, height: 50, toJSON: jest.fn()}
		model.setBox(rect);

		expect((model.box)).toEqual(rect);
	})

})