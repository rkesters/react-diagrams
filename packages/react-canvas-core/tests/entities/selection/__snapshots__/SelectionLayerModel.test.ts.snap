// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionLayerModel Constructor 1`] = `
Object {
  "id": Any<String>,
  "isSvg": false,
  "transformed": false,
  "type": "selection",
}
`;
